src/XMakeBuildEngine/Xml/ProjectXmlUtilities.cs

```diff
@@ -48,6 +48,13 @@ private static List<XmlElementWithLocation> GetChildElements(XmlElementWithLocat
                         VerifyThrowProjectValidNamespace(childElement);
                         children.Add(childElement);
                         break;
+                    case XmlNodeType.Text:
+                        // Whitespace is read as a #text element so if the node is purely whitespace, just ignore it.
```

Comment:
It's odd that we're getting a whitespace-only text node when it should come back as a Whitespace node. But the reason is that there's a limit in `XmlTextReaderImpl`'s determine-node-type code: if it's [greater than 4kB of whitespace](https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImpl.cs#L5571-L5586), it'll return as `Text`.

But given that it works this way, this check sounds good.